icu4c/source/tools/toolutil/pkg_genc.cpp

```diff
@@ -1254,11 +1257,14 @@ writeObjectCode(
     uprv_memset(&symbolNames, 0, sizeof(symbolNames));
 
     /* write the linker export directive */
-    uprv_strcpy(objHeader.linkerOptions, "-export:");
+    if (optSkipDllExport == NULL)
```

Comment:
It's boolean, so
  if (!optSkipDllExport) {
    ...

Reply:
Done.